pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -899,11 +905,17 @@ impl ImageLayerWriterInner {
         let desc = PersistentLayerDesc::new_img(
             self.tenant_shard_id,
             self.timeline_id,
-            self.key_range.clone(),
+            if let Some(end_key) = end_key {
+                self.key_range.start..end_key
+            } else {
+                self.key_range.clone()
+            },
             self.lsn,
             metadata.len(),
         );
 
+        // TODO: when end_key is specified, verify that the end key is in range
```

Comment:
can you implement this TODO? Maybe via an assert or error log or something?

Reply:
This requires we copy the last-written key somewhere when `put_image` is called, and then verify it at the end. I plan to add that in a separate pull request with `#[cfg(test)]`.

Reply:
added in https://github.com/neondatabase/neon/pull/8574/commits/bb60bdfa68fa6028518607940b5a52e63a0e9b29, turned out to be a small change